src/main/java/teammates/ui/controller/ControllerServlet.java

```diff
@@ -124,18 +121,12 @@ public final void doPost(HttpServletRequest req, HttpServletResponse resp) throw
                 resp.sendRedirect(Const.ViewURIs.ERROR_PAGE);
             }
         } catch (Throwable t) {
-            String requestMethod = req.getMethod();
-            String requestUserAgent = req.getHeader("User-Agent");
-            String requestPath = req.getServletPath();
-            String requestUrl = req.getRequestURL().toString();
-            String requestParams = HttpRequestHelper.printRequestParameters(req);
-
-            EmailWrapper errorReport =
-                    new EmailGenerator().generateSystemErrorEmail(requestMethod, requestUserAgent, requestPath,
-                                                                  requestUrl, requestParams, userType, t);
-            new EmailSender().sendReport(errorReport);
-            log.severe(ActivityLogEntry.generateSystemErrorReportLogMessage(req, errorReport, userType));
-
+            /*
+             * Log only basic info about the exception to prevent delay in termination of request which can result
+             * in GAE shutting down the instance.
+             * Note that severe logs are sent by email automatically in the cron job auto/compileLogs.
+             */
+            log.severe("Unexpected exception " + t.getMessage());
```

Comment:
Good work. While you are at it, see if there is a way to include the stack trace in the log message. Use `TeammatesException#toStringWithStackTrace(...)`?